### PR TITLE
refactor: rewrite several extension ops to op2

### PR DESCRIPTION
**Pull request by bartlomieju**
Rewrites following extensions:
- `ext/web`
- `ext/url`
- `ext/webstorage`
- `ext/io`

### CODE REVIEW

File: ext/url/lib.rs [hunk 1]
```diff
@@ -148,13 +152,14 @@ fn as_u32_slice(slice: &mut [u8]) -> &mut [u32] {
...
+#[op2(fast)]
+#[smi]
 pub fn op_url_reparse(
   state: &mut OpState,
-  href: String,
-  setter: u8,
-  setter_value: String,
-  buf: &mut [u8],
+  #[string] href: String,
+  #[smi] setter: u8,
```
Comment (mmastrac) [inline ext/url/lib.rs:160]: You can omit #[smi] here.

File: ext/web/lib.rs [hunk 2]
```diff
@@ -414,6 +425,7 @@ fn op_encoding_encode_into(
   out_buf[1] = boundary as u32;
 }
 
+// TODO(bartlomieju): op2 doesn't support v8::ArrayBuffer
```
Comment (mmastrac) [inline ext/web/lib.rs:428]: It should. Does a `&mut v8::ArrayBuffer` work?

Comment (mmastrac) [reply ext/web/lib.rs:428]: This should work:

```
#[op2]
fn op_transfer_arraybuffer<'a>(
  scope: &mut v8::HandleScope<'a>,
  ab: &v8::ArrayBuffer,
) -> Result<v8::Local<'a, v8::ArrayBuffer>, AnyError> {
  if !ab.is_detachable() {
    return Err(type_error("ArrayBuffer is not detachable"));
  }
  let bs = ab.get_backing_store();
  ab.detach(None);
  let ab = v8::ArrayBuffer::with_backing_store(scope, &bs);
  Ok(ab)
}
```

File: ext/url/lib.rs [hunk 3]
```diff
@@ -148,13 +152,14 @@ fn as_u32_slice(slice: &mut [u8]) -> &mut [u32] {
...
+  #[string] setter_value: String,
+  #[buffer] buf: &mut [u8],
```
Comment (mmastrac) [inline ext/url/lib.rs:162]: This is actually a legit bug -- the buffer is Uint32Array on the JS side.

Comment (bartlomieju) [reply ext/url/lib.rs:162]: Ah, nice catch, that's probably why the tests are failing.

File: ext/web/lib.rs (hunk 2 above)
Comment (bartlomieju) [reply ext/web/lib.rs]: This indeed works!

File: ext/url/lib.rs (hunk 3 above)
Comment (mmastrac) [reply ext/url/lib.rs:162]: Updated to main -- will try again

**Pull request merged**